consistency/tool/clustering/clustering.py

```diff
@@ -0,0 +1,114 @@
+
+from cmath import nan
+import numpy as np
+import pandas as pd
+from sklearn.cluster import dbscan
+from strsimpy.sorensen_dice import SorensenDice
+from sklearn.cluster import dbscan
+from word_extraction import ExtractWords
+from collections import Counter
+import consistency.rpc.google.cloud.apigeeregistry.applications.v1alpha1.consistency.word_group_pb2 as wg
+
+from google.protobuf.json_format import ParseDict
+class ClusterWords:
+    def __init__(self, stub):
+      self.stub = stub
+
+    def get_words(self):
+        extrct = ExtractWords(stub="stub")
+        words = extrct.get_vocabs()
+
+        if words is None:
+            return None
+        return words
+
+    def clean_words(self):
+        words = self.get_words()
+
+        if words is None:
+            return None
+        for word in words:
+            if type(word) != str or len(word) < 2 or word != word:
+                words.remove(word)
+        words_length = len(words)
+
+        if words_length < 3:
+            print("Only ", words_length, " words found. Forming clusters not possible.")
+            return None
+
+        return words
+
+    def cluster(self):
+
+        words = self.clean_words()
+
+        if words is None:
+            return words
+
+        def extract_indices_dice(x, y):
+
+            dice = SorensenDice(2)
+            nonlocal words
+            i, j = int(x[0]), int(y[0])  
+            return dice.distance(words[i], words[j])
+
+        data = np.arange(len(words)).reshape(-1, 1)
+
+        try:
+            db = dbscan(data, metric=extract_indices_dice, eps=0.3, min_samples=2, algorithm='brute')
+        except Exception as e:
```

Comment:
Along with the exception, what if dbscan returns all `-1` response? (The ones we saw in examples where it was not able to converge) we should handle that too.